simplify `updating.md` (#1803)
meilisearch/documentation: learn/advanced/updating.md

# Update to the latest Meilisearch version

> :::warning

Author: I don't like how this looks:

<img width="780" alt="Screen Shot 2022-08-08 at 16 40 05" src="https://user-images.githubusercontent.com/90181761/183420144-922ab6ff-7e3e-49f1-9b07-8b71fbd425a4.png">

But I'm not sure what would be a better solution

Reviewer: I think this should be split in two warnings. 

The first (guide  does not work on <v0.15) is in the right place. Perhaps we could move it to the end of the intro—might be more aesthetically pleasing not to have the warning right next to the page title. You could also use `::: danger` instead of `::: warning` if you want to be extra scary.

The second warning (auth header) could perhaps be leave where it was? Or perhaps just as the last line on the intro, as regular text? If you think that the problem is the code sample taking too much space, I think we could do without it.

1. This guide does not work for versions below v0.15. For more information, contact [support](https://slack.meilisearch.com/).
2. If you're using v0.24 or below, use the `X-MEILI-API-KEY: apiKey` authorization header:

 <CodeSamples id="updating_guide_check_version_old_authorization_header" />
:::

Currently, Meilisearch databases can only be opened by the Meilisearch version you used to create them. The following guide will walk you through all the steps to migrate an existing database from an older version of Meilisearch to the most recent one.

If you already have a Meilisearch database with some data you don’t want to lose, you are in the right place!

:::note
If you have already installed the latest version and manually indexed your data and settings, you can ignore this guide.

> :::

Reviewer: Do we really need this note? Isn't it essentially saying "if you have already updated meilisearch, you don't need to follow this guide on updating meilisearch"?

If you get the error `Cannot open database, expected Meilisearch engine version: 0.X.X, current engine version 0.Y.Y`, your database is not compatible with the currently installed Meilisearch version.

> In this case, you need to download the compatible version now (i.e., `0.X.X` in the above error message) to access and export your database.

Reviewer: Great edit 👍

If you are updating to v0.28, keys imported from the old version will have their `key` and `uid` fields regenerated.
:::

## Step 2: Set all fields as displayed attributes

::: note
If your dump was created in Meilisearch v0.21 or above, continue to [step 3](#step-3-create-the-dump).
:::

When creating dumps, Meilisearch calls the same method as the [get documents endpoint](/reference/api/documents.md#get-documents). This means that all fields must be [displayed](/learn/configuration/displayed_searchable_attributes.md#displayed-fields) in order to be saved in the dump.

> Start by using the get displayed attributes endpoint to verify that all attributes are displayed:

Reviewer: If the link is not necessary, I think we can make this sentence sound a bit more natural:

```suggestion
Start by verifying that all attributes are included in the displayed attributes list:
```

Or something to that effect. I find the `the [something] endpoint` construct a bit wooden when we're not using it to link to the reference.

<CodeSamples id="updating_guide_get_displayed_attributes_new" />

If the response is `{'displayedAttributes': '["*"]'}`, you can move on to the [next step](#step-3-create-the-dump).

> If it's something else, then you need to use the reset displayed attributes endpoint. Before doing this, make sure you save your list of displayed attributes somewhere so you can restore it afterwards.

Reviewer: ```suggestion
If it's something else, then you need to reset the list of displayed attributes. Before doing this, make sure you save your list of displayed attributes somewhere so you can restore it afterwards.
```

::::

> To create a dump, use the create dump endpoint:

Reviewer: ```suggestion
You can then create a dump of your database:
```
